pykrylov/linop/linop.py

```diff
@@ -720,6 +720,45 @@ def matvec_transp(y):
                           matvec_transp=matvec_transp, symmetric=symmetric)
 
 
+def CysparseLinearOperator(A):
+    """
+    Return a linear operator from a CySparse sparse matrix.
+
+    Note: it is intentional that we do not use CySparse syntaxic sugar to avoid
+          having interferences with CySparse proxies. Thus we prefer `A.matvec`,
+          `A.matvec_transp` and `A.matvec_htransp` to `A * x` `A.T * x` and
+          `A.H * x`.
+    """
+
+    nargout, nargin = A.shape
+    symmetric = A.is_symmetric
+
+    def matvec(x):
+        if x.shape != (nargin,):
+            msg = 'Input has shape ' + str(x.shape)
+            msg += ' instead of (%d,)' % nargin
+            raise ShapeError(msg)
+        return A.matvec(x)
```

Comment:
Does `CySparse` already perform an argument check? If so, there's no need to perform it twice.

Also, can't we write `A * x`?